sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -1252,28 +1257,31 @@ int getRetryTimeoutInMs(int retry) {
         return (int) options.getRetryTimeout().apply(Duration.ofSeconds(retry)).toMillis();
     }
 
-    private Mono<Boolean> checkIMDSAvailable(String endpoint) {
-        return Mono.fromCallable(() -> {
-            HttpURLConnection connection = null;
-            URL url = getUrl(endpoint + "?api-version=2018-02-01");
+    Mono<Boolean> checkIMDSAvailable(String endpoint) {
 
+        URL url = null;
+        try {
+            url = getUrl(endpoint + "?api-version=2018-02-01");
+        } catch (MalformedURLException e) {
+            throw new RuntimeException(e);
+        }
+        HttpClient client = HttpClient.createDefault(new HttpClientOptions().setConnectTimeout(Duration.ofSeconds(1)));
+        HttpRequest request = new HttpRequest(HttpMethod.GET, url);
```

Comment:
Is this a probe request or a fully formed token request?

Reply:
Just a probe. I'll rename the method and add a doc too.

Reply:
We should test ACI with this change.